crates/compression/src/decompress.rs

```diff
@@ -58,7 +81,15 @@ pub async fn decompress<D>(
 where
     D: DecompressDb,
 {
-    // TODO: merkle root verification: https://github.com/FuelLabs/fuel-core/issues/2232
+    #[cfg(feature = "fault-proving")]
+    {
+        let db_registry_root = db.registry_root()?;
+        if let VersionedCompressedBlock::V1(block) = &block {
+            if block.header.registry_root != db_registry_root {
+                anyhow::bail!("Registry root mismatch");
+            }
+        }
```

Comment:
It would be nice to have match statement here to handle new version in the future here.

Reply:
addressed in https://github.com/FuelLabs/fuel-core/commit/f7668148d1cf80496867845a15a9c0a2e17cbac2#diff-d0b271dacdbc77c5808aab48fee0fa85f338d593e1adc25e8bd0205c370dd824R86-R94